examples/features/metadata_interceptor/server/main.go

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Binary server is an example server.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"math/rand"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	pb "google.golang.org/grpc/examples/features/proto/echo"
+)
+
+var port = flag.Int("port", 50051, "the port to serve on")
+
+var errMissingMetadata = status.Errorf(codes.InvalidArgument, "missing metadata")
+
+type server struct {
+	pb.UnimplementedEchoServer
+}
+
+func unaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, errMissingMetadata
+	}
+
+	md.Append("key1", "value1")
+	ctx = metadata.NewIncomingContext(ctx, md)
+
+	return handler(ctx, req)
+}
+
+func (s *server) UnaryEcho(ctx context.Context, in *pb.EchoRequest) (*pb.EchoResponse, error) {
+	fmt.Printf("--- UnaryEcho ---\n")
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return nil, status.Errorf(codes.DataLoss, "UnaryEcho: failed to get metadata")
+	}
+
+	// Read metadata from interceptor
+	if v, ok := md["key1"]; ok {
+		fmt.Printf("key1 from metadata: \n")
+		for i, e := range v {
+			fmt.Printf(" %d. %s\n", i, e)
+		}
+	}
+
+	fmt.Printf("request received: %v, sending echo\n", in)
+
+	return &pb.EchoResponse{Message: in.Message}, nil
+}
+
+type wrappedStream struct {
+	grpc.ServerStream
+	ctx context.Context
+}
+
+func (s *wrappedStream) Context() context.Context {
+	return s.ctx
+}
+
+func streamInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	md, ok := metadata.FromIncomingContext(ss.Context())
+	if !ok {
+		return errMissingMetadata
+	}
+
+	md.Append("key1", "value1")
+	ctx := metadata.NewIncomingContext(ss.Context(), md)
+
+	return handler(srv, &wrappedStream{ss, ctx})
+}
+
+func (s *server) BidirectionalStreamingEcho(stream pb.Echo_BidirectionalStreamingEchoServer) error {
+	fmt.Printf("--- BidirectionalStreamingEcho ---\n")
+
+	md, ok := metadata.FromIncomingContext(stream.Context())
+	if !ok {
+		return status.Errorf(codes.DataLoss, "BidirectionalStreamingEcho: failed to get metadata")
+	}
+
+	// Read metadata from interceptor.
+	if v, ok := md["key1"]; ok {
+		fmt.Printf("key1 from metadata: \n")
+		for i, e := range v {
+			fmt.Printf(" %d. %s\n", i, e)
+		}
+	}
+
+	// Read requests and send responses.
+	for {
+		in, err := stream.Recv()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		fmt.Printf("request received %v, sending echo\n", in)
+		if err = stream.Send(&pb.EchoResponse{Message: in.Message}); err != nil {
+			return err
+		}
+	}
+}
+
+func main() {
+	flag.Parse()
+	rand.Seed(time.Now().UnixNano())
```

Comment:
This seems to be unused. Could you please remove it.

Reply:
Got it, I will fix it.